core/src/test/java/io/questdb/test/cairo/ArrayTest.java

```diff
@@ -245,6 +245,25 @@ public void testArrayCanBeClearedAfterInstantiation() throws Exception {
         });
     }
 
+    @Test
+    public void testArrayFirstFunction() throws Exception {
+        assertMemoryLeak(() -> {
+            execute("create table test (ts timestamp, x int, v double[]) timestamp(ts) partition by DAY");
+            execute("insert into test(ts,x,v) values ('2022-02-24', 1, ARRAY[1.0,1.0]), ('2022-02-24', 2, null), ('2022-02-24', 3, ARRAY[2.0,2.0])");
+
+            assertQuery(
+                    "ts\tx\tv\n" +
+                            "2022-02-24T00:00:00.000000Z\t1\t[1.0,1.0]\n" +
+                            "2022-02-24T00:00:00.000000Z\t2\tnull\n" +
+                            "2022-02-24T00:00:00.000000Z\t3\t[2.0,2.0]\n",
+                    "select ts, x, first(v) as v from test sample by 1s",
```

Comment:
do you mind also asserting execution plan to ensure it tests the merge? and also the number of threads executing the SQL

Reply:
ah, i see this is done in other tests